test/models/test_albef.py

```diff
@@ -4,113 +4,220 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
+from functools import partial
+
+import pytest
 import torch
 from test.test_utils import assert_expected, set_rng_seed
 from torch import nn, Tensor
 from torchmultimodal.models.albef import ALBEFModel, ALBEFSimilarity
+from torchmultimodal.modules.encoders.albef_multimodal_encoder import (
+    ALBEFMultimodalEncoder,
+)
+from torchmultimodal.modules.encoders.albef_text_encoder import ALBEFTextEncoder
+from torchmultimodal.modules.encoders.albef_vision_encoder import ALBEFVisionEncoder
+
+
+@pytest.fixture(autouse=True)
+def vision_encoder():
+    set_rng_seed(0)
+    return ALBEFVisionEncoder(
+        image_size=4,
+        patch_size=4,
+        num_layers=2,
+        num_heads=1,
+        hidden_dim=3,
+        mlp_dim=6,
+        norm_layer=partial(nn.LayerNorm, eps=1e-6),
+    )
+
+
+@pytest.fixture(autouse=True)
+def text_encoder():
+    return ALBEFTextEncoder(hidden_size=3, num_attention_heads=1)
 
 
-class TestALBEFModel:
-    albef = ALBEFModel(
+@pytest.fixture(autouse=True)
+def multimodal_encoder():
+    return ALBEFMultimodalEncoder(hidden_size=3, num_attention_heads=1)
+
+
+@pytest.fixture(autouse=True)
+def dummy_albef_model():
+    return ALBEFModel(
+        nn.Linear(3, 2),
         nn.Linear(3, 2),
         nn.Linear(3, 2),
         nn.Linear(3, 2),
         nn.Linear(3, 2),
+        embed_dim=2,
+        queue_size=4,
+    )
+
+
+@pytest.fixture(autouse=True)
+def albef_model(vision_encoder, text_encoder, multimodal_encoder):
```

Comment:
I thought autouse meant you don't have to add any of these as args? (Sorry, this may just be my noob pytest understanding...)

Reply:
I still need to pass them in as args (or else `albef_model` refers to the function, not what's returned in the function)